webui/src/pages/repositories/repository/objects.jsx

```diff
@@ -3,22 +3,232 @@ import React, {useRef, useState} from "react";
+                    <p>Import succeeded!</p>
+                </div>
+                <div style={{display: 'flex', justifyContent: 'center'}}>
+                    <p><strong><div style={{ color: 'green', display: 'inline'}}> {numObjects} </div></strong> objects imported and committed into branch <strong>{importBranch}</strong></p>
+                </div>
+                <div style={{display: 'flex', justifyContent: 'center'}}>
+                    <p> &nbsp;<Link to={`compare?ref=${ currBranch }&compare=${ importBranch }`} variant = "success" className="btn btn-success">Compare and Merge</Link>&nbsp;it into the current branch - <strong>{currBranch}</strong></p>
+                </div>
+                <Alert variant="warning" className={"ml-2 mr-2 row mt-4"}>
+                    Merging {importBranch} will delete all files on branch {currBranch}
+                </Alert>
+
+            </Col>
+        </Row>
+    );
+}
+
+
+const ImportButton = ({ config, repo, reference, path, onDone, onClick, variant = "success", onHide, show = false, enabled= false}) => {
+    const initialState = {
+        inProgress: false,
+        error: null,
+        done: false,
+        numObj: 0,
+        isStorageNamespaceValid: null
+    }
+    const [importState, setImportState] = useState(initialState)
+
+    const sourceRef = useRef(null);
+    const destRef = useRef(null);
+    const commitMsgRef = useRef(null);
+    const storageType = config.blockstore_type
+    const DEFAULT_BLOCKSTORE_VALIDITY_REGEX = new RegExp(`^s3://`);
+    const storageNamespaceValidityRegexStr = config ? config.blockstore_namespace_ValidityRegex : DEFAULT_BLOCKSTORE_VALIDITY_REGEX;
+    const storageNamespaceValidityRegex = RegExp(storageNamespaceValidityRegexStr);
+    const sourceURIExample = config ? config.blockstore_namespace_example : "s3://my-bucket/path/";
+
+
+    if (!reference || reference.type !== RefTypeBranch) return <></>
+
+    const checkStorageNamespaceValidity = () => {
+        const isStorageNamespaceValid = storageNamespaceValidityRegex.test(sourceRef.current.value)
+        setImportState({inProgress: importState.inProgress, error: importState.error, done: importState.done, numObj: importState.numObj, isStorageNamespaceValid: isStorageNamespaceValid});
+    };
+
+    const hide = () => {
+        if (importState.inProgress) return;
+        setImportState(initialState)
+        onHide()
+    };
+
+    const ingest = async () => {
+        setImportState({
+            ...initialState,
+            inProgress: true
+        })
+
+        try {
+            let hasMore = true
+            let continuationToken = ""
+            let after = ""
+            let prepend = `${destRef.current.value}`;
+            let importBranchResp
+            let sum = importState.numObj
+            const importBranch = `${reference.id}-imported`;
```

Comment:
Suggestion:
```suggestion
            const importBranch = `_${reference.id}_imported`;
```

This way, the new branch:
1. Looks more "internal".
2. Lower chance of colliding with existing branch protection rules. For example: consider "stable/1.0.0" as the import destination and a branch protection rule with the pattern "stable/*". With the current code, the import will fail because "stable/1.0.0-imported" is protected. Of course, it's not bullet proof - if you have a rule of "*stable*" this doesn't work.

Reply:
Done, thanks for the suggestion!